File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.size() == 0) {
            return 0;
        }
        int n = needle.size();
        int h = haystack.size();
        if (n > h) {
            return -1;
        }
        int i = 0;
        bool found = true;
        int k;
        while (i <= (h-n)) {
            found = true;
            k = i;
            while ((found) && (k < (i+n))) {
                if (haystack[k] != needle[k-i]) {
                    found = false;
                }
                k++;
            }
            if (found) {
                return i;
            }
            i++;
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        if (needle.len() == 0) {
            return 0;
        }
        let haystack_vec: Vec<char> = haystack.chars().collect();
        let needle_vec: Vec<char> = needle.chars().collect();
        let n: usize = needle.len() as usize;
        let h: usize = haystack.len() as usize;
        if (n > h) {
            return -1;
        }
        let mut i: usize = 0;
        let mut found: bool = true;
        while (i <= (h-n)) {
            found = true;
            for k in i..(i+n) as usize {
                if (haystack_vec[k] != needle_vec[k-i]) {
                    found = false;
                    break;
                }
            }
            if (found) {
                return i as i32;
            }
            i += 1;
        }
        return -1;
    }
}